File: Disciplinas/AED/AED2/verde/palindromo/com-funcao/Main.java
import java.util.Scanner;

class Main {

    public static boolean ehPalindromo(String str) {
        boolean palin = true;
        int n = str.length();
        
        for(int i = 0; i <= n/2; i++) {
            if(str.charAt(i) != str.charAt(n-i-1)) {
                palin = false;
                i = n;
            }
        }

        return palin;
    }

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        String input;

        do{
            input = scan.nextLine();

            if(!input.equals("FIM")) {
                if(ehPalindromo(input)) System.out.println("SIM");
                else System.out.println("NAO");
            }
        }while(!input.equals("FIM"));

        scan.close();
    }
}

File: Disciplinas/AED/AED2/verde/palindromo/com-funcao/main.c
#include <stdio.h>
#include <string.h>

int ehPalindromo(char* str, int n) {
        int palin = 1;

        for(int i = 0; i <= n/2; i++) {
                if(str[i] != str[n-i-1]) {
                        palin = 0;
                        i = n;
                }
        }

        return palin;
}

void charEspecial(char* str, int n) {
        for(int i = 0; i < n; i++) {
                if(str[i] < 32 || str[i] > 127) str[i] = '.';
        }
} 


int main(void) {

        char input[1000];

        do{
                fgets(input,1000,stdin);
                input[strcspn(input,"\n")] = '\0';
                input[strcspn(input,"\r")] = '\0';
                int tam = strlen(input);

                if(strcmp(input, "FIM") != 0) {
                        charEspecial(input, tam);
                        if(ehPalindromo(input, tam)) printf("SIM\n");
                        else printf("NAO\n");
                }

        }while(strcmp(input, "FIM") != 0);

        return 0;
}
